### PR TITLE
MAKEFILE: Auto-generate dependencies for scummvm.rc

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by orgads** [comment]
Missing: `$(srcdir)/` prefix for most deps, glue for actually using the d file, and removal of the existing deps in win32.mk.

Pro: Auto-detected
Con: Lists more than actually used. For example, if an engine is excluded, the rc file will still depend on its resources.

**Comment by lephilousophe** [comment]
I think this PR is better than #2919

File: Makefile.common [hunk 1]
```diff
@@ -203,9 +203,13 @@ endif
 	$(QUIET_AS)$(AS) $(ASFLAGS) $(<) -o $@
 
 # Build rule for Windows resource files
-# TODO: Support dependency tracking
+dists/scummvm.rc: config.mk
+
 %.o: %.rc
 	$(QUIET)$(MKDIR) $(*D)
+	$(QUIET)$(MKDIR) $(*D)/$(DEPDIR)
+	$(QUIET)echo -n "$@: $< " > $(*D)/$(DEPDIR)/scummvm.d
```
Comment (digitall) [inline Makefile.common:211]: Given the complexity of this line and the following sed magic, I would suggest a comment to explain what this is intended to do.

Comment (digitall) [reply Makefile.common:211]: i.e. I have written a fair amount of sed scripts, but that level is a bit XKCD 208 :)

Comment (orgads) [reply Makefile.common:211]: 😆

**Comment by orgads** [comment]
Ready for review.

**Comment by orgads** [comment]
Done. Ready for review.

**Comment by orgads** [comment]
@digitall?

**Comment by digitall** [comment]
Looks reasonable. The comments you added to explain the sed magic look fine. Have not tested here as not on Windows, maybe @ccawley2011 should review?

File: Makefile.common [hunk 2]
```diff
@@ -203,9 +203,17 @@ endif
...
+# The regular expression is a portable form of ^(.*\s(FILE|ICON)|#include)\s+"([^"]*)".*$
+# This finds lines in the form '... FILE "path/file.ext" ...' (and also ICON and #include)
+# and replaces them with the $(srcdir)/path/file.ext \
+# The second sed removes winresrc.h (system include) and config.h (not in srcdir, printed
+# by the echo before sed), and strips the trailing backslash on the last line.
 %.o: %.rc
 	$(QUIET)$(MKDIR) $(*D)
+	$(QUIET)$(MKDIR) $(*D)/$(DEPDIR)
+	$(QUIET)echo "$@: $< config.h config.mk \\" > $(*D)/$(DEPDIR)/scummvm.d
+	$(QUIET)sed -n 's:^\(.*[[:space:]]\(FILE\|ICON\)\|#include\)[[:space:]][[:space:]]*"\([^"]*\)".*$$:    $(srcdir)/\3 \\:p; ' $(<) | \
```
Comment (ccawley2011) [inline Makefile.common:215]: It would be useful to detect lines containing ``RT_MANIFEST`` and ``DATA`` as well.

**Review by ccawley2011 — COMMENTED** [review]
Looks alright, just one minor comment.

File: Makefile.common (hunk 2 above)
Comment (orgads) [reply Makefile.common]: Done

**Comment by orgads** [comment]
ping?

**Comment by sev-** [comment]
Thanks, makes sense. Good work. Merging.

**Pull request merged**